### ISSUE TITLE
`ZodDiscriminatedUnionDef` broke in `zod` 3.20.0

### ISSUE THREAD

**Issue by freben** [StefanTerdell/zod-to-json-schema#31]
Zod 3.20.0 was just released, and while it claims to have no breaking changes, `ZodDiscriminatedUnionDef` now only has two type arguments. Therefore, builds using `zod-to-json-schema` now fail with

```
node_modules/zod-to-json-schema/src/parsers/union.d.ts:22:58 - error TS2314: Generic type 'ZodDiscriminatedUnionDef<Discriminator, Options>' requires 2 type argument(s).

22 export declare function parseUnionDef(def: ZodUnionDef | ZodDiscriminatedUnionDef<any, any, any>, refs: References): JsonSchema7PrimitiveUnionType | JsonSchema7AnyOfType | undefined;
```

**Comment by colinhacks** [comment]
Thanks for letting me know, I'll add a default to that generic which should resolve that issue. I export literally everything from `types.ts` so people have an easier time building things on top of Zod. The downside is that there is virtually no change I can make that won't be "breaking" if someone relies on a particular type alias. For that reason, I consider the _public API_ to be the `z.<whatever>`, the method signatures, and the internal structure of ZodError - the things that a typical user will interact with.  But historically I haven't considered the structure of Defs or the generic type signatures of each subclass part of the public API, hence "no breaking changes". I'll take more care to minimize changes to the internals to account for the growing ecosystem of tools that rely on Zod.

**Comment by StefanTerdell** [comment]
@colinhacks fwiw I stand by this approach

anyway, this is fixed with zod-to-json-schema@3.20.0 so closing
